pkg/chains/formats/intotoite6/pipelinerun/pipelinerun.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pipelinerun
+
+import (
+	"time"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/tektoncd/chains/pkg/chains/formats/intotoite6/attest"
+	"github.com/tektoncd/chains/pkg/chains/formats/intotoite6/extract"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/apis"
+)
+
+const (
+	TektonPipelineRunID = "https://tekton.dev/attestations/chains/pipelinerun@v2"
+)
+
+type BuildConfig struct {
+	Tasks []TaskAttestation `json:"tasks"`
+}
+
+type TaskAttestation struct {
+	Name       string                    `json:"name,omitempty"`
+	After      []string                  `json:"after,omitempty"`
+	Ref        v1beta1.TaskRef           `json:"ref,omitempty"`
+	StartedOn  time.Time                 `json:"startedOn,omitempty"`
+	FinishedOn time.Time                 `json:"finishedOn,omitempty"`
+	Status     string                    `json:"status,omitempty"`
+	Steps      []attest.StepAttestation  `json:"steps,omitempty"`
+	Invocation slsa.ProvenanceInvocation `json:"invocation,omitempty"`
+	Results    []v1beta1.TaskRunResult   `json:"results,omitempty"`
+}
```

Comment:
Maybe out of scope of this pr. 
I think we could consolidate `Ref` and `Invocation` once https://github.com/tektoncd/chains/pull/554 and related work in pipeline repo are done. Specifically, we can remote `Ref` field and just record remote refs in `TaskAttestation.Invocation.configSource`. Happy to take on this part after this PR is merged if we agree on this.

Reply:
Yeah I agree, that would be great